analyses/snv-callers/scripts/03-calculate_tmb.R

```diff
@@ -143,46 +141,109 @@ if (!dir.exists(opt$output)) {
   dir.create(opt$output, recursive = TRUE)
 }
 
-# Declare output file paths
-tmb_coding_file <- file.path(opt$output, "pbta-snv-consensus-mutation-tmb-coding.tsv")
-tmb_all_file <- file.path(opt$output, "pbta-snv-consensus-mutation-tmb-all.tsv")
+# Get data name
+data_name <- ifelse(opt$tcga, "tcga", "pbta")
+
+# Declare output file based on data_name
+tmb_coding_file <- file.path(
+  opt$output,
+  paste0(data_name, "-snv-mutation-tmb-coding.tsv")
+)
+tmb_all_file <- file.path(
+  opt$output,
+  paste0(data_name, "-snv-mutation-tmb-all.tsv")
+)
 
 # Don't bother if both files exist already and overwrite is FALSE
 if (all(file.exists(c(tmb_coding_file, tmb_all_file)), !opt$overwrite)) {
   stop(paste0(tmb_coding_file, tmb_all_file, "both exist and --overwrite is not
               being used. Use --overwrite if you would like to overwrite these files."))
 }
-########################### Set up this consensus data ##########################
-# Print progress message
-message(paste("Reading in", opt$consensus, "MAF data..."))
 
-# Read in this MAF, skip the version number
-maf_df <- data.table::fread(opt$consensus, data.table = FALSE)
+######################## Obtain Mutect Strelka mutations #######################
+# Start up connection
+con <- DBI::dbConnect(RSQLite::SQLite(), opt$db_file)
+
+# Designate caller tables from SQL file
+strelka <- dplyr::tbl(con, "strelka")
+mutect <- dplyr::tbl(con, "mutect")
+
+# Specify the columns to join by
+join_cols <- c(
+  "Chromosome",
+  "Start_Position",
+  "Reference_Allele",
+  "Allele",
+  "Tumor_Sample_Barcode"
+)
+
+# Create the consensus for non-MNVs
+strelka_mutect_maf_df <- strelka %>%
+  dplyr::inner_join(mutect, by = join_cols)
+
+# Get Multi-nucleotide calls from mutect as SNVs
+split_mutect_df <- split_mnv(mutect)
+
+# join MNV calls with strelka
+strelka_mutect_mnv <- strelka %>%
+  dplyr::inner_join(split_mutect_df,
+    by = join_cols,
+    copy = TRUE
+  ) %>%
+  as.data.frame()
+
+if (opt$tcga) {
+  strelka_mutect_mnv <- strelka_mutect_mnv %>%
+    dplyr::mutate(Tumor_Sample_Barcode = substr(Tumor_Sample_Barcode, 0, 12))
```

Comment:
What is going on here? Can you add a comment?

Reply:
In the TCGA MAF files, the Tumor_Sample_Barcode has the biospecimen information but only the first 12 characters are used for matching to the the metadata